packages/core/src/lib/connectors.ts

```diff
@@ -0,0 +1,19 @@
+import { connectorInstances, getConnectorInstanceById } from '@/connectors';
+import { ConnectorInfo } from '@/connectors/types';
+import { findConnectorById } from '@/queries/connector';
+
+export const getAllConnectorInfo = async (): Promise<ConnectorInfo[]> => {
+  return Promise.all(
+    connectorInstances.map(async (connectorInstance) => {
+      const { metadata } = connectorInstance;
+      const connector = await findConnectorById(metadata.id);
+      return { ...connector, metadata };
+    })
+  );
+};
+
+export const getConnectorInfoById = async (id: string): Promise<ConnectorInfo> => {
```

Comment:
1. one file but two data sources, we can just use `ConnectorInstance` as the SSOT
2. if so, i don't think we still need `ConnectorInfo` type, we can organize it as below:

```ts
export interface ConnectorInstance {
  connector: Connector; // this holds the db entry
  metadata: ConnectorMetadata;
  sendMessage: EmailSendMessageFunction;
  validateConfig: ValidateConfig;
}
```

and we just trim the instance in api response